src/main/scala/nl/knaw/dans/api/sword2/DepositHandler.scala

```diff
@@ -160,6 +166,38 @@ object DepositHandler {
     }
   }
 
+  private def resolveFetchItems(bagitDir: File)(implicit id: String): Try[Unit] = Try {
+    log.debug(s"[$id] Checking fetch.txt")
+    getFetchTxt(bagitDir).foreach(fetchText =>
+      { log.debug(s"[$id] Resolving files in fetch.txt")
+        fetchText.asScala.foreach(item =>
+        getUrl(item.getUrl).foreach(url =>
+          Using.urlInputStream(url).foreach(src =>
+            Files.copy(src, Paths.get(bagitDir.getAbsolutePath, item.getFilename)))))
+      }
+    )
+  }
+
+  private def getFetchTxt(bagitDir: File)(implicit id: String): Option[FetchTxt] = Option {
+    getBagFromDir(bagitDir).getFetchTxt
+  }
+
+  private def getBagFromDir(dir: File)(implicit id: String): Bag = {
+    try {
+      bagFactory.createBag(dir, BagFactory.Version.V0_97, BagFactory.LoadOption.BY_MANIFESTS)
+    } catch {
+      case e: Throwable => throw new FailedDepositException(id, "Failed to create a bag when resolving Fetch Items", e)
+    }
+  }
+
+  private def getUrl(url: String)(implicit id: String): Try[URL] =  {
+    val urlPattern = SwordProps("fetch.allowed-url-pattern").r
+    urlPattern findFirstIn url match {
+      case Some(_) => Success(new URL(url))
```

Comment:
The name of the method tells exactly what it does. It just isn't what we really want to do. I think the following is better:

    if (urlPattern.pattern.matcher(url).matches) // Success(new URI(url)) 
    else // Failure(...)

Although it is a bit more verbose, it expresses and does what we need: match the complete URL-string against the configured pattern. If we choose the wrong `urlPattern` we would get unexpected results with the code as it is now. Compare:

     "http://.*".r.pattern.matcher("not a valid url http://www.google.com").matches() // return false
     "http://.*".r.findFirstIn("not a valid url http://www.google.com").isDefined // returns true

Since we didn't put in the starting "^" and closing "$" we get an incorrect result in the second line.